### ISSUE TITLE
List of measurements contains no elements

### ISSUE THREAD

**Issue by Symbai** [dotnet/BenchmarkDotNet#953]
I've created a new console application targeting .NET Framework 4.7.2. Added the example code on github page. Added BenchmarkDotNet 0.11.2 via Nuget. Switched to Release build and run.

Got this exception thrown, nothing was benchmarked:

```
System.InvalidOperationException: List of measurements contains no elements
   at BenchmarkDotNet.Extensions.ReportExtensions.GetStatistics(IReadOnlyCollection`1 runs)
   at BenchmarkDotNet.Analysers.ZeroMeasurementAnalyser.<AnalyseReport>d__5.MoveNext()
   at BenchmarkDotNet.Analysers.AnalyserBase.<Analyse>d__2.MoveNext()
   at System.Linq.Enumerable.<SelectManyIterator>d__17`2.MoveNext()
   at System.Linq.Enumerable.<DistinctIterator>d__64`1.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at BenchmarkDotNet.Running.BenchmarkRunner.PrintSummary(ILogger logger, IConfig config, Summary summary)
   at BenchmarkDotNet.Running.BenchmarkRunner.Run(BenchmarkRunInfo[] benchmarkRunInfos, IConfig commonSettingsConfig)
   at BenchmarkDotNet.Running.BenchmarkRunner.Run(BenchmarkRunInfo benchmarkRunInfo)
   at BenchmarkDotNet.Running.BenchmarkRunner.Run[T](IConfig config)
```

The console window shows lots of errors, such as:

```
File not found: System.Runtime, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
....
 error CS0234: The type or namespace name 'Characteristics' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) 
```

I can run the example code myself without BenchmarkDotNet without issues.

**Comment by Rizzen** [comment]
@Symbai It's a known issue and will be fixed in 0.11.3. For now, you can use package from [bdn-nightly feed](https://benchmarkdotnet.org/articles/guides/nuget.html#nightly)

**Comment by Symbai** [comment]
I've tried the latest package 0.11.2.849 from nightly feed. The benchmark no longer crashes with an exception but nothing got benchmark. The log is full of something like this:

```
// Build Exception: The file or assembly "System.Runtime, Version = 4.1.2.0, Culture = neutral, PublicKeyToken = b03f5f7f11d50a3a" or a dependency of it was not found. The system can not find the stated file.

// Build Exception: The file or assembly "System.Runtime, Version = 4.1.2.0, Culture = neutral, PublicKeyToken = b03f5f7f11d50a3a" or a dependency of it was not found. The system can not find the stated file.

Error during the build process.

C: \ Program Files \ dotnet \ sdk \ 2.1.403 \ Microsoft.Common.CurrentVersion.targets (2110,5): warning MSB3277: Conflicts have been found between different versions of "System.Collections.Immutable" that will not be resolved could. These reference conflicts are listed in the build log if the log precision is set to Verbose.

 Method |    Job | Runtime |     N | Mean | Error | Ratio | RatioSD | Rank |
------- |------- |-------- |------ |-----:|------:|------:|--------:|-----:|
 Sha256 |    Clr |     Clr |  1000 |   NA |    NA |     ? |       ? |    ? |
 Sha256 |   Core |    Core |  1000 |   NA |    NA |     ? |       ? |    ? |
 Sha256 | CoreRT |  CoreRT |  1000 |   NA |    NA |     ? |       ? |    ? |
        |        |         |       |      |       |       |         |      |
    Md5 |    Clr |     Clr |  1000 |   NA |    NA |     ? |       ? |    ? |
    Md5 |   Core |    Core |  1000 |   NA |    NA |     ? |       ? |    ? |
    Md5 | CoreRT |  CoreRT |  1000 |   NA |    NA |     ? |       ? |    ? |
        |        |         |       |      |       |       |         |      |
 Sha256 |    Clr |     Clr | 10000 |   NA |    NA |     ? |       ? |    ? |
 Sha256 |   Core |    Core | 10000 |   NA |    NA |     ? |       ? |    ? |
 Sha256 | CoreRT |  CoreRT | 10000 |   NA |    NA |     ? |       ? |    ? |
        |        |         |       |      |       |       |         |      |
    Md5 |    Clr |     Clr | 10000 |   NA |    NA |     ? |       ? |    ? |
    Md5 |   Core |    Core | 10000 |   NA |    NA |     ? |       ? |    ? |
    Md5 | CoreRT |  CoreRT | 10000 |   NA |    NA |     ? |       ? |    ? |

Benchmarks with issues:
  Md5VsSha256.Sha256: Clr(Runtime=Clr) [N=1000]
  Md5VsSha256.Sha256: Core(Runtime=Core) [N=1000]
  Md5VsSha256.Sha256: CoreRT(Runtime=CoreRT) [N=1000]
  Md5VsSha256.Md5: Clr(Runtime=Clr) [N=1000]
  Md5VsSha256.Md5: Core(Runtime=Core) [N=1000]
  Md5VsSha256.Md5: CoreRT(Runtime=CoreRT) [N=1000]
  Md5VsSha256.Sha256: Clr(Runtime=Clr) [N=10000]
  Md5VsSha256.Sha256: Core(Runtime=Core) [N=10000]
  Md5VsSha256.Sha256: CoreRT(Runtime=CoreRT) [N=10000]
  Md5VsSha256.Md5: Clr(Runtime=Clr) [N=10000]
  Md5VsSha256.Md5: Core(Runtime=Core) [N=10000]
  Md5VsSha256.Md5: CoreRT(Runtime=CoreRT) [N=10000]
```

**Comment by AndreyAkinshin** [comment]
@Symbai it looks similar to #896. Could you try to delete app.config file?

**Comment by Symbai** [comment]
@AndreyAkinshin that at least removes the System.Runtime not found from log. But using 0.11.2.851 from nightly still won't let me benchmark anything.

Log contains lots of these errors:

```
14cfca6f-d0e2-42c8-b046-d8a88284b272.notcs(13,23): error CS0234: The type or namespace name 'Characteristics' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [C:\Users\User\source\repos\ConsoleApp1123\bin\Release\14cfca6f-d0e2-42c8-b046-d8a88284b272\BenchmarkDotNet.Autogenerated.csproj]
14cfca6f-d0e2-42c8-b046-d8a88284b272.notcs(14,23): error CS0234: The type or namespace name 'Environments' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [C:\Users\User\source\repos\ConsoleApp1123\bin\Release\14cfca6f-d0e2-42c8-b046-d8a88284b272\BenchmarkDotNet.Autogenerated.csproj]
14cfca6f-d0e2-42c8-b046-d8a88284b272.notcs(15,23): error CS0234: The type or namespace name 'Jobs' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [C:\Users\User\source\repos\ConsoleApp1123\bin\Release\14cfca6f-d0e2-42c8-b046-d8a88284b272\BenchmarkDotNet.Autogenerated.csproj]
14cfca6f-d0e2-42c8-b046-d8a88284b272.notcs(16,23): error CS0234: The type or namespace name 'Extensions' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [C:\Users\User\source\repos\ConsoleApp1123\bin\Release\14cfca6f-d0e2-42c8-b046-d8a88284b272\BenchmarkDotNet.Autogenerated.csproj]
```

but also a few times this here

```
// Build Exception: Standard output:
   Packages for "C: \ Users \ User \ source \ repos \ ConsoleApp1123 \ bin \ Release \ 66d1d7eb-71e3-4c1f-b028-8d310dd3cc0b \ BenchmarkDotNet.Autogenerated.csproj" are restored ...
  FindPackagesByIdAsyncCore is repeated for the source https://ci.appveyor.com/nuget/benchmarkdotnet/FindPackagesById()?id=Microsoft.DotNet.ILCompiler'&semVerLevel=2.0.0.
  Access to a socket was illegal due to the access rights of the socket
  FindPackagesByIdAsyncCore is repeated for the source https://ci.appveyor.com/nuget/benchmarkdotnet/FindPackagesById()?id=Microsoft.DotNet.ILCompiler'&semVerLevel=2.0.0.
  Access to a socket was illegal due to the access rights of the socket
  "FindPackagesByIdAsync" is repeated for the source "https://api.nuget.org/v3-flatcontainer/microsoft.dotnet.ilcompiler/index.json".
  Access to a socket was illegal due to the access rights of the socket
  "FindPackagesByIdAsync" is repeated for the source "https://api.nuget.org/v3-flatcontainer/microsoft.dotnet.ilcompiler/index.json".
  Access to a socket was illegal due to the access rights of the socket
C: \ Program Files \ dotnet \ sdk \ 2.1.403 \ NuGet.targets (114.5): error: The service index for the source "https://dotnet.myget.org/F/dotnet-core/api/v3 /index.json "could not be loaded. [C: \ Users \ User \ source \ repos \ ConsoleApp1123 \ bin \ Release \ 66d1d7eb-71e3-4c1f-b028-8d310dd3cc0b \ BenchmarkDotNet.Autogenerated.csproj]
C: \ Program Files \ dotnet \ sdk \ 2.1.403 \ NuGet.targets (114,5): error: The access to a socket was not allowed due to the access rights of the socket [C: \ Users \ User \ source \ repos \ ConsoleApp1123 \ bin \ release \ 66d1d7eb-71e3-4c1f-b028-8d310dd3cc0b \ BenchmarkDotNet.Autogenerated.csproj]
```

Also tried making a completely new console project, disabled "Auto-generate binding redirects" and add nightly nuget package but its still the same.

**Comment by adamsitnik** [comment]
The sample code from GitHub targets 4 different frameworks: Clr, Mono, Core and CoreRT.

Your project should target both .NET and .NET Core to make it work: `<TargetFrameworks>net4720;netcoreapp2.1</TargetFrameworks>`

If you want to just benchmark .NET 4.7.2 you need to remove the `MonoJob`, `CoreJob` and `CoreRtJob` attributes:

```diff
- [ClrJob(baseline: true), CoreJob, MonoJob, CoreRtJob]
+ [ClrJob]
```

**Comment by Symbai** [comment]
@adamsitnik Thanks. After I did that, VS informed me that .NET core SDK was missing. So I went to dot.net and installed latest version of both. After installation the benchmark finally succeed.

Then I removed that line you mentioned and it still succeed except that core benchmark was skipped (but no issues or exception like it has happened before). So the problem was core SDK missing only. Plus your line to get core benchmarks result. I think it would be great mentioning the requirements and recommended settings on github or in log. Maybe add a check that if core sdk is missing the benchmark doesn't crash.
